---
title: "mmrf"
author: "Anand Srinivasan"
date: "27/04/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(MMRFBiolinks)
library(TCGAbiolinks)
library(SummarizedExperiment)
library(dplyr)
library(DT)
```

```{r}
therapy<-"Bortezomib" 
tsample<-"TBM"
clin<-MMRFGDC_QueryClinic(type = "clinical")
sample.type = c("Primary Blood Derived Cancer - Bone Marrow")

query.mm <- GDCquery(
  project = "MMRF-COMMPASS",
  data.category = "Transcriptome Profiling",
  data.type = "Gene Expression Quantification",
  experimental.strategy = "RNA-Seq",
  workflow.type = "STAR - Counts",
  sample.type = sample.type
)

GDCdownload(query.mm)

MMRnaseqSE <- GDCprepare(query.mm,
                              save = TRUE ,
                              save.filename = "MM_RNASeqSE.rda",
                              directory = "GDCdata",
                              summarizedExperiment = TRUE, 
                              #mut.pipeline = NULL,
                              #mutant_variant_classification = NULL,
                              add.gistic2.mut = NULL)



getResults(query.mm, cols = c("cases.submitter_id","sample_type","cases")) %>% datatable(options = list(scrollX = TRUE, keys = TRUE))




#MMRnaseqSE <- ("MM_RNASeqSE.rda")
colData(MMRnaseqSE) %>% as.data.frame() %>% 
  filter(patient %in% ID_samples) %>% View()


IDs_Bort<-MMRFGDC_QuerySamples(query=query.mm,typesample=tsample, clin.mm=clin,therapyname=therapy) #case 1



clinMMGateway_2 <- read.delim("MMRF_CoMMpass_IA17_STAND_ALONE_TRTRESP_V2.tsv") %>%
  rename("public_id" = 1)


first_line_Bort <-
  clinMMGateway_2 %>% 
  filter(trtclass == "Bortezomib-based") %>%
  filter(trtstdy ==1 ) %>%
  filter(line==1) %>%
  filter(public_id %in% IDs_Bort)

ID_samples <- first_line_Bort %>%
  select(public_id, response) %>%
  distinct()

TBM_Bort_samples_SE <- MMRnaseqSE[, MMRnaseqSE$patient %in% ID_samples]

TBM_Bort_samples_SE_colData <- colData(TBM_Bort_samples_SE) %>% 
  data.frame() %>%
  left_join(ID_samples, 
  by = c("patient" = "public_id")) 

a <- 
  SummarizedExperiment(
  assays = assay(TBM_Bort_samples_SE),
  rowData = rowData(TBM_Bort_samples_SE),
  #rowRanges = rowRanges(TBM_Bort_samples_SE),
  colData = TBM_Bort_samples_SE_colData
)
  
write.table(assay(a) %>% 
              as.data.frame() %>% 
              rownames_to_column("id") %>% 
              mutate(id = tools::file_path_sans_ext(id)),
                paste0("genes_MMRF.tsv"),
                col.name=TRUE,
                sep="\t",
                na = "NA",
                row.names=FALSE,
                quote=FALSE)

colData(a) %>% as.data.frame() %>% dplyr::select(response) %>% rownames_to_column("SampleName") %>%
  mutate(response = ifelse(response == 1, "Responder", "Non-responder")) %>%
  write.table(paste0("design_MMRF.tsv"),
                col.name=TRUE,
                sep="\t",
                na = "NA",
                row.names=FALSE,
                quote=FALSE)






```

